FeedDetails.js

```diff
@@ -174,6 +174,7 @@ class FeedDetails {
    * Incorporates the given FeedDetails into this one, updating our properties.
    */
   update(feedDetails) {
+    this.parent = feedDetails.parent || this.parent
```

Comment:
Instead of closing this PR, we could change this line to

```suggestion
    this.parent = this.parent || feedDetails.parent
```

Then I'd be much more comfortable. Because we don't want a subfeed to mutate its parent, that would be weird. But if it's null/undefined, then we do want to set it to a valid parent.

Reply:
Right, thanks. Will merge after the tests pass